modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -3941,7 +3927,7 @@ else if (msg instanceof SnapshotFilesFailureMessage) {
         }
 
         /** {@inheritDoc} */
-        @Override public void onException(UUID nodeId, Throwable ex) {
+        @Override public synchronized void onException(UUID nodeId, Throwable ex) {
```

Comment:
I think we don't need `RemoteSnapshotFilesRecevier task = active;` any more

Reply:
Fixed